### ISSUE TITLE
No map visible because the map container's width or height are 0

### ISSUE THREAD

**Issue by johndeighan** [MelihAltintas/vue3-openlayers#210]
I'm getting this warning with OpenLayers 7.4.0, even though I've set the container div to have a width and height via CSS. Then, the map appears, but the copyright message is incorrectly placed. Here is my web page:

```
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="utf-8"/>
	<title>My First Parcel App</title>
	<style>
		#map {
			width: 800px;
			height: 600px;
			}
	</style>
</head>
<body>

<h1>Open Layers Tutorial</h1>

<div id="map"></div>

<!-- --------------------------------------------- -->
<script type="module">

import View from 'ol/View.js';
import Map from 'ol/Map.js';
import OSM from 'ol/source/OSM.js';
import TileLayer from 'ol/layer/Tile.js';
import {defaults} from 'ol/control/defaults';
import FullScreen from 'ol/control/FullScreen.js';
import MousePosition from 'ol/control/MousePosition.js';
import OverviewMap from 'ol/control/OverviewMap.js';

let targetElem = document.getElementById('map');

let view = new View({
	center: [0,0],
	zoom: 2,
	minZoom: 2,
	maxZoom: 6,
	enableRotation: true,
	rotation: 0,
	});
let map = new Map({
	target: targetElem,
	view,
	layers: [
		new TileLayer({
			source: new OSM()
			})
		],
	keyboardEventTarget: document,
	controls: defaults().extend([
		new FullScreen(),
		new MousePosition(),
		new OverviewMap(),
		]),
	});
console.log("Working");

</script>

</body>
</html>

```

Though I doubt it's relevant, I'm using parcel for my dev server since it supports hot reloading. I previously got the very same behavior when using SvelteKit, but there's a lot of complexity there so I wanted to do the same with a simple web server. FYI, I install the Open Layers library using `
/`

**Expected behavior**
Should get no warning. Furthermore, though the copyright message is placed part way down the page, i.e. not at the bottom, any mouse clicks above the copyright message don't work, i.e. can't be captured via an on click handler.

**Screenshots**
There's no explanation of how to add a screen shot, nor an item in the above menu for that

**Desktop (please complete the following information):**

- OS: Windows 11
- Browser Firefox, Chrome (latest versions)
- Version of Open Layers: 7.4.0

**Comment by d-koppenhagen** [comment]
@johndeighan your code refers to Openlayers itself and not to this Vue3 specific wrapper library. Please raise an issues there or check out the existing one.
https://github.com/openlayers/openlayers

Furthermore "No map visible because the map container's width or height are 0" has already been answered here: https://github.com/MelihAltintas/vue3-openlayers/issues/22#issuecomment-1622569687

**Comment by johndeighan** [comment]
How can you close this issue??? I can also search for the issue and the one you reference has solutions that simply don't work, plus that's a very old issue with an old version. Also, this has absolutely nothing to do with Vue3 - I have no idea where you got that from. I don't think you even looked at my code.

**Comment by d-koppenhagen** [comment]
Exactly, this seems to have nothing to do with vue, your code in the description is plain OpenLayers, so please open an issue in the Openlayers repo. This Project simply wraps the Openlayers API, meaning a problem that occurs in Openlayers will also occur here. Nonetheless, the references issue has also been closed as there was a solution by defineing the width/height using CSS.
In your code above, you are importing all Classes directly from Openlayers `ol/*`. So you don't use anything from this library.

If you think your problem still belongs to this project, please:

- provide example code, that shows your actual problem, ideally using the provided [Stackblitz starter template](https://stackblitz.com/edit/vue3-openlayers-template)
- re-open this issue and describe the problem in detail with the related code

**Comment by johndeighan** [comment]
Update:

1. This was my mistake. I did a web search for the error message I was receiving, came to this page, and didn't notice that it was specific to Vue3. Sorry about that. I'll post the issue on the main openlayers page.
2. The problem that I was having, even with a very simple web page using OpenLayers, was due to not linking to the OpenLayers CSS file. I'm now doing that with `<link rel="stylesheet" href="../node_modules/ol/ol.css">` in my `<head>` section. Note that I've installed OpenLayers with `npm install ol`, so this is specific to NodeJS. There's probably a better way to link to the stylesheet - this way won't work if they change the placement of the `ol.css` file inside the `node_modules `folder.
3. The error message I was receiving is completely bogus. I had set the height and width of the map container, but due to the missing link to ol.css, it wasn't being recognized. That threw me off.
